sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTestBase.java

```diff
@@ -488,4 +510,64 @@ static void assertRestException(Throwable exception, Class<? extends HttpRespons
             assertEquals(exception, ex.getClass());
         }
     }
+
+    /**
+     * Helper method to verify that two configuration setting are equal. Users can defined their equal method.
+     *
+     * @param o1 ConfigurationSetting object 1
+     * @param o2 ConfigurationSetting object 2
+     * @return boolean value that defines if two ConfigurationSettings are equal
+     */
+    boolean equals(ConfigurationSetting o1, ConfigurationSetting o2) {
+        if (o1 == o2) {
+            return true;
+        }
+
+        if (!Objects.equals(o1.getKey(), o2.getKey())
+            || !Objects.equals(o1.getLabel(), o2.getLabel())
+            || !Objects.equals(o1.getValue(), o2.getValue())
+            || !Objects.equals(o1.getETag(), o2.getETag())
+            || !Objects.equals(o1.getLastModified(), o2.getLastModified())
+            || !Objects.equals(o1.isLocked(), o2.isLocked())
+            || !Objects.equals(o1.getContentType(), o2.getContentType())
+            || ImplUtils.isNullOrEmpty(o1.getTags()) != ImplUtils.isNullOrEmpty(o2.getTags())) {
+            return false;
+        }
+
+        if (!ImplUtils.isNullOrEmpty(o1.getTags())) {
+            return Objects.equals(o1.getTags(), o2.getTags());
+        }
+
+        return true;
+    }
+
+    /**
+     * A helper method to verify that two lists of ConfigurationSetting are equal each other.
+     *
+     * @param settings1 List of ConfigurationSetting
+     * @param settings2 Another List of ConfigurationSetting
+     * @return boolean value that defines if two ConfigurationSetting lists are equal
+     */
+    boolean equalsArray(List<ConfigurationSetting> settings1, List<ConfigurationSetting> settings2) {
+        if (settings1 == settings2) {
+            return true;
+        }
+
+        if (settings1 == null || settings2 == null) {
+            return false;
+        }
+
+        if (settings1.size() != settings2.size()) {
+            return false;
+        }
+
+        final int size = settings1.size();
+        for (int i = 0; i < size; i++) {
```

Comment:
What if the contents aren't ordered in the same way? Will that affect this equality?

Reply:
It is list, so i think the order matters.

Reply:
Does the REST API guarantee an order that the configuration settings are returned in?

Reply:
Not sure what the REST API guaranteed applying to this method.  But I can explain what I think of. This equal method is a helper function that could be one of many equals() method users want to define how is equality of two settings. It just could be one of them.